hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/node/HapiNodeCreate.java

```diff
@@ -0,0 +1,157 @@
+        this.accountId = Optional.of(accountID);
+        return this;
+    }
+
+    public HapiNodeCreate addDesc(final String description) {
+        this.description = description;
+        return this;
+    }
+
+    public HapiNodeCreate addGossipEndpoint(final List<ServiceEndpoint> gossipEndpoint) {
+        this.gossipEndpoint = gossipEndpoint;
+        return this;
+    }
+
+
+
+    public HapiNodeCreate addServiceEndpoint(final List<ServiceEndpoint> serviceEndpoint) {
+        this.serviceEndpoint = serviceEndpoint;
+        return this;
+    }
+
+    public HapiNodeCreate addGossipCaCertificate(final byte[] gossipCaCertificate) {
+        this.gossipCaCertificate = gossipCaCertificate;
+        return this;
+    }
+
+    public HapiNodeCreate addGrpcCertificateHash(final byte[] grpcCertificateHash) {
+        this.grpcCertificateHash = grpcCertificateHash;
+        return this;
+    }
+
+    @Override
+    protected HapiNodeCreate self() {
+        return this;
+    }
+
+    @Override
+    protected long feeFor(final HapiSpec spec, final Transaction txn, final int numPayerKeys) throws Throwable {
+        return spec.fees().forActivityBasedOp(HederaFunctionality.CryptoCreate, this::usageEstimate, txn, numPayerKeys);
+    }
+
+    private FeeData usageEstimate(final TransactionBody txn, final SigValueObj svo) {
+        //temp till we decide about the logic
+        return FeeData.newBuilder().setNodedata(FeeComponents.newBuilder().setBpr(0)).setNetworkdata(FeeComponents.newBuilder().setBpr(0)).setServicedata(FeeComponents.newBuilder().setBpr(0)).build();
+    }
+
+    @Override
+    protected Consumer<TransactionBody.Builder> opBodyDef(final HapiSpec spec) throws Throwable {
+        final NodeCreateTransactionBody opBody = spec.txns()
+                .<NodeCreateTransactionBody, NodeCreateTransactionBody.Builder>body(
+                        NodeCreateTransactionBody.class, b -> {
+                            if(accountId.isPresent())
+                                b.setAccountId(accountId.get());
+                            b.setDescription(description);
+                            b.addAllGossipEndpoint(gossipEndpoint);
+                            b.addAllServiceEndpoint(serviceEndpoint);
+                            b.setGossipCaCertificate(ByteString.copyFrom(gossipCaCertificate));
+                            b.setGrpcCertificateHash(ByteString.copyFrom(grpcCertificateHash));
+                        });
+        return b -> b.setNodeCreate(opBody);
+    }
+
+    @Override
+    protected List<Function<HapiSpec, Key>> defaultSigners() {
+        return Arrays.asList(spec -> spec.registry().getKey(effectivePayer(spec)));
+    }
+
+    @Override
+    protected void updateStateOf(final HapiSpec spec) {
+        final var newId = lastReceipt.getNodeId();
+//        newNumObserver.ifPresent(obs -> obs.accept(newId.getFileNum());
+        spec.registry().saveNodeId(description, newId);
```

Comment:
We should save `NodeId` with `AccountID`. Not with description. description might be duplicated

Reply:
Fixed to use nodeName instead of description